h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -381,6 +381,11 @@ to a map (possibly the metadata map) -
                     storeLock.lock();
                     try {
                         readStoreHeader();
+                    } catch (final OutOfMemoryError oom) {
```

Comment:
Why do you think that OOM is limited to that call only? It could potentially be anywhere, and database will not be closed, so we are back to square one. What exactly may cause OOM in real run, as oppose to the one you are faking in this test?